taxcalc/utils.py

```diff
@@ -19,6 +21,7 @@
 except ImportError:
     BOKEH_AVAILABLE = False
 
+TAXCALC_PATH = os.path.dirname(os.path.realpath(__file__))
```

Comment:
Hey @talumbau @martinholmer, sorry for the delay on this one. `TAXCALC_PATH` should always evaluate to the true (resolving symlinks) path to the taxcalc directory, in my case:

```
>>> taxcalc.utils.TAXCALC_PATH
'c:\\Sites\\ospc\\tax-calculator\\taxcalc'
```


There's quite a few places within the codebase where we use a direct child directory: 

```
CUR_PATH = os.path.abspath(os.path.dirname(__file__))
```
 in `validation/csv_in.py`, `tests/conftest.py`, 4 files in `taxbrain/`, and 2 files in `comparison/`.

But this exact string is only used twice within the taxcalc package:

* `CUR_PATH = os.path.abspath(os.path.dirname(__file__))` in `records.py`
* `path = os.path.join(os.path.abspath(os.path.dirname(__file__)), cls.DEFAULTS_FILENAME)` in `parameters.py`

My original idea was that we could define this in `taxcalc.utils`, only access the filesystem to resolve the directory once (slightly improving load times?), and improve code reuse by importing it and referencing it directly, e.g.

```
from taxcalc.utils import TAXCALC_PATH

...

path = os.path.join(TAXCALC_PATH, cls.DEFAULTS_FILENAME)
```

in `parameters.py` or 

```
TAXBRAIN_PATH = os.path.join(TAXCALC_PATH, 'taxbrain')
```

at the top of `taxbrain`, and then each of its 4 files would reference `TAXBRAIN_PATH`.

But I didn't realize that none of these sub directories are actually set up as sub-packages of `taxcalc` (using a `__init__.py` files). I'm also not sure if the file structure of these in relation to `taxcalc` changes when it's installed via a package manager, which is I believe what @talumbau was suggesting.

So there's my use case - if it's worth fixing, let me know @talumbau. Otherwise I will drop it from this PR.